release/cut-official-release.sh

```diff
@@ -0,0 +1,338 @@
+#!/bin/bash
+
+# Copyright 2015 The Kubernetes Authors All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script automates the release processes for all pre-releases and official
+# releases for Kubernetes.  See docs/devel/releasing.md for more info.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+# Sets DIR, INSTRUCTIONS
+function main() {
+  # Parse arguments
+  if [[ "$#" -ne 2 ]]; then
+    usage
+    exit 1
+  fi
+  local -r new_version=${1-}
+  # TODO(ihmccreery) Stop calling it githash; it's not a githash.
+  local -r githash=${2-}
+  DRY_RUN=true
+  if [[ "${3-}" == "--no-dry-run" ]]; then
+    echo "!!! This NOT is a dry run."
+    DRY_RUN=false
+  else
+    echo "This is a dry run."
+  fi
+
+  check-prereqs
+
+  # Get and verify version info
+  local -r alpha_version_regex="^v(0|[1-9][0-9]*)\\.(0|[1-9][0-9]*)\\.0-alpha\\.([1-9][0-9]*)$"
+  local -r beta_version_regex="^v(0|[1-9][0-9]*)\\.(0|[1-9][0-9]*)\\.(0|[1-9][0-9]*)-beta\\.([1-9][0-9]*)$"
+  local -r official_version_regex="^v(0|[1-9][0-9]*)\\.(0|[1-9][0-9]*)\\.(0|[1-9][0-9]*)$"
+  local -r series_version_regex="^v(0|[1-9][0-9]*)\\.(0|[1-9][0-9]*)$"
+  if [[ "${new_version}" =~ $alpha_version_regex ]]; then
+    local -r release_type='alpha'
+    local -r version_major="${BASH_REMATCH[1]}"
+    local -r version_minor="${BASH_REMATCH[2]}"
+    local -r version_alpha_rev="${BASH_REMATCH[3]}"
+  elif [[ "${new_version}" =~ $beta_version_regex ]]; then
+    local -r release_type='beta'
+    local -r version_major="${BASH_REMATCH[1]}"
+    local -r version_minor="${BASH_REMATCH[2]}"
+    local -r version_patch="${BASH_REMATCH[3]}"
+    local -r version_beta_rev="${BASH_REMATCH[4]}"
+  elif [[ "${new_version}" =~ $official_version_regex ]]; then
+    local -r release_type='official'
+    local -r version_major="${BASH_REMATCH[1]}"
+    local -r version_minor="${BASH_REMATCH[2]}"
+    local -r version_patch="${BASH_REMATCH[3]}"
```

Comment:
nit: Can you perhaps remove some of the cut 'n paste code for release_type, version_major, version_minor, version_alpha_rev here?  For example by lifting common code out to above the 'if' statement?  Same comment applies to the regex's above.  They are all almost identical.  Can you factor out the common stuff?